lms/static/js/spec/learner_dashboard/program_details_view_spec.js

```diff
@@ -603,6 +603,27 @@ define([
             expect($(view.$('.select-choice')[0]).attr('for')).toEqual($(view.$('.run-select')[0]).attr('id'));
             expect($(view.$('.enroll-button button')[0]).text().trim()).toEqual('Enroll Now');
         });
+
+        it('should send analytic event when purchase button clicked', function() {
+            var properties = {
+                category: 'partial bundle',
+                label: 'Test Course Title',
+                uuid: '0ffff5d6-0177-4690-9a48-aa2fecf94610'
+            };
+            view = initView({
+                programData: $.extend({}, options.programData, {
+                    is_learner_eligible_for_one_click_purchase: true,
+                    variant: 'partial'
+                })
+            });
+            view.render();
+            $('.upgrade-button').click();
```

Comment:
Future proofing: Did you consider adding .complete-program to this click as there is another button on the page that uses the class upgrade-button for a single course run?